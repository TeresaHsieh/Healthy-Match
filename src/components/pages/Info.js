import React from "react";
import { connect } from "react-redux";
import { Redirect } from "react-router-dom";
import "../../css/info.css";
import ChefMatch from "../../imgs/chef-match.png";
import Edit from "../../imgs/edit.png";
import BasicInfo from "../../components/others/ComOfInfo/BasicInfo";
import AuthInfo from "../../components/others/ComOfInfo/AuthInfo";
import ContributionInfo from "../others/ComOfInfo/ContributionInfo";
import { checkUserInfo } from "../../store/actions/authAction";
import SignUpRedirect from "../others/ComOfMember/SignUpRedirect";

import LoveMatch from "../../imgs/love-match.png";

// App Components
import Header from "../common/Header";
import { relativeTimeRounding } from "moment";

class Info extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      openFirstSignUpBox: false
    };
  }

  componentDidMount = () => {
    // this.props.checkUserInfo(this.props.auth.uid);
    if (
      // !this.props.userInfo.Name &&
      !this.props.userInfo.Age &&
      !this.props.userInfo.Weight &&
      !this.props.userInfo.Height &&
      !this.props.userInfo.Sexual
    ) {
      this.setState({ openFirstSignUpBox: true });
    }
  };

  handleInfoChange = () => {
    console.log("調整資料");
  };

  closeFirstSignUpBox = () => {
    this.setState({ openFirstSignUpBox: false });
  };

  render() {
    const { auth } = this.props;
    if (!auth.uid) return <Redirect to="./member" />;

    const openFirstSignUpBox = this.state.openFirstSignUpBox;

    return (
      <div>
        <Header />
        <div className="main-info">
          {openFirstSignUpBox ? (
            <div className="backgroundSignUpRedirectBox">
              <div className="signUpRedirectBox">
                <SignUpRedirect />{" "}
                <button onClick={this.closeFirstSignUpBox}>我知道了！</button>
              </div>
            </div>
          ) : (
            ""
          )}
          <div className="simple-info">
            {/* <h1>
              HELLO :)
              <br /> {this.props.userInfo.Name}
            </h1> */}
            {/* <AuthInfo /> */}
          </div>
          <div className="more-info">
            <BasicInfo />
            <ContributionInfo />
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    date: new Date().toLocaleDateString(),
    auth: state.firebase.auth,
    userInfo: state.firebase.profile
  };
};

const mapDispatchToProps = dispatch => {
  return {
    // create a method
    checkUserInfo: userUID => {
      dispatch(checkUserInfo(userUID));
    }
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Info);
