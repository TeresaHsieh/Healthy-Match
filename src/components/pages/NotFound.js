import React from "react";
import "../../css/404.css";

// App Components
import Header from "../common/Header";
import SadMatch from "../../imgs/sad-match.png";

const NotFound = () => (
  <div>
    <Header />
    <div className="main-404">
      <div>
        <h2 className="code404"> 404 </h2>
        <h2> 喔不！此頁不存在！ </h2>
        <p> 請再次確認網址路徑～麻吉感謝你 ❤️ </p>
      </div>
      <img src={SadMatch} className="sad-match" />
    </div>
  </div>
);

export default NotFound;
