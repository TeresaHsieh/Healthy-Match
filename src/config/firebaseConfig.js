import firebase from "firebase/app";
import "firebase/firestore";
import "firebase/auth";

// Initialize Firebase
var firebaseConfig = {
  apiKey: "AIzaSyAu5M8SOdRdZjRp6HwRB9LqdkdUyNTvNlg",
  authDomain: "healthy-match.firebaseapp.com",
  databaseURL: "https://healthy-match.firebaseio.com",
  projectId: "healthy-match",
  storageBucket: "healthy-match.appspot.com",
  messagingSenderId: "560759415450",
  appId: "1:560759415450:web:9fc855be631767dc"
};

firebase.initializeApp(firebaseConfig);
firebase.firestore().settings({ timestampsInSnapshots: true });

export default firebase;
export const auth = firebase.auth();
export const googleProvider = new firebase.auth.GoogleAuthProvider();
export const facebookProvider = new firebase.auth.FacebookAuthProvider();
export const emailProvider = new firebase.auth.EmailAuthProvider();

// module.exports = firebaseConfig;
