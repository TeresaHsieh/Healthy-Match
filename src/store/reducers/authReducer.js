const initState = {
  authError: null
};
const authReducer = (state = initState, action) => {
  switch (action.type) {
    case "LOGIN_ERROR":
      console.log(action.err.message);
      return {
        ...state,
        authError: "Login Failed",
        loginError: action.err.message
      };

    case "LOGIN_SUCCESS":
      return {
        ...state,
        authError: null
      };

    case "SIGNOUT_SUCCESS":
      return state;

    case "SIGNUP_SUCCESS":
      return {
        ...state,
        authError: null
      };

    case "SIGNUP_ERROR":
      console.log(action.err.message);
      return {
        ...state,
        authError: action.err.message,
        signupError: action.err.message
      };

    case "CHECK_USER_INFO":
      return {
        ...state,
        userInfo: action.UserInfo
      };

    case "UPDATE_INFO_TO_FIRESTORE":
      return {
        ...state,
        userInfo: action.wholeState
      };

    case "SENT_LAST_IMG_TO_REDUX_STORE":
      return {
        ...state,
        LastIMG: action.LastIMG
      };

    case "SENT_DESCRIPTION_TO_REDUX_STORE":
      return {
        ...state,
        description: action.stateDescription
      };

    case "GET_CONTRIBUTION_DETAILS":
      return {
        ...state,
        contributionDetails: action.contributionDetail
      };

    case "REMOVE_IMG_AND_DESCRIPTION":
      console.log("進到reducer");
      return {
        ...state,
        LastIMG: undefined,
        description: undefined
      };

    default:
      return state;
  }
};
export default authReducer;
