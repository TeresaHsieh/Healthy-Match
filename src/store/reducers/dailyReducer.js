// import * as ActionTypes from "../actions/actionTypes";

const initState = {
  date: new Date().toLocaleDateString(),
  meals: "breakfast" // status default is breakfast
};

const dailyReducer = (state = initState, action) => {
  let meal = state.meals;
  switch (action.type) {
    case "UPDATE_DAILY_RECORDS_NAME":
      return {
        ...state,
        recordName: action.newRecord
      };
    case "UPDATE_DAILY_RECORDS_SERVE":
      return {
        ...state,
        recordServe: action.newRecord
      };
    // case "ADD_RECORD_INPUT":
    //   console.log(action.inputName);
    //   console.log(action.inputServe);

    //   console.log(state);
    //   let meal = state.meals;
    //   return {
    //     ...state,
    //     recordName: state.recordName[meal].concat(action.inputName),
    //     recordServe: state.recordServe[meal].concat(action.inputServe)
    //   };

    case "ADD_RECORD_INPUT_NAME":
      return {
        ...state,
        recordName: state.recordName.concat(action.nextInputName)
        // recordName: state.recordName[meal].concat(action.nextInputName)
        // recordName: state.recordName[meal].splice(0, 0, action.nextInputName)
      };
    case "ADD_RECORD_INPUT_SERVE":
      return {
        ...state,
        recordServe: state.recordServe.concat(action.nextInputServe)
      };

    case "SEND_DATA_TO_FIREBASE":
      return {
        ...state,
        recordName: state.recordName,
        recordServe: state.recordServe
      };
    case "ADD_RECORD_INPUT_ERR":
      console.log("add input error", action.err);
      return state;
    default:
      return state;
  }
};

export default dailyReducer;
