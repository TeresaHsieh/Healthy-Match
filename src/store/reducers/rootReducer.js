import memberReducer from "./memberReducer";
import dailyReducer from "./dailyReducer";
import { firestoreReducer } from "redux-firestore"; // sync firestore data to our state

import { combineReducers } from "redux";

const rootReducer = combineReducers({
  member: memberReducer,
  daily: dailyReducer,
  firestore: firestoreReducer
});

export default rootReducer;
