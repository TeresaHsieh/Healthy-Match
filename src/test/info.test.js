import React from "react";
import { BrowserRouter, Route, Switch } from "react-router-dom";
import { firebaseReducer } from "react-redux-firebase";
import { combineReducers } from "redux";
import { cleanup, fireEvent, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { createStore, applyMiddleware, compose } from "redux";
import { Provider } from "react-redux";
import { exportAllDeclaration } from "@babel/types";
import rootReducer from "../store/reducers/rootReducer";
import dailyReducer from "../store/reducers/dailyReducer";
import authReducer from "../store/reducers/authReducer";
import Info from "../components/pages/Info";
import NotFound from "../components/pages/NotFound";

import ReactDOM from "react-dom";
import thunk from "redux-thunk";
import { reduxFirestore, getFirestore } from "redux-firestore";
import { reactReduxFirebase, getFirebase } from "react-redux-firebase";
import firebaseConfig from "../config/firebaseConfig";

test("Test Info", () => {
  let store = createStore(rootReducer);
  const infoWithRudex = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Info />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(infoWithRudex).toMatchSnapshot();
});

test("Test Info with UID", () => {
  const initState = {
    // store: { auth: { uid: "r097WikN8kfJbZkWlZItqw7wMx73" } },
    // auth: { uid: "r097WikN8kfJbZkWlZItqw7wMx73" },
    firebase: {
      auth: {
        isLoaded: false,
        isEmpty: true,
        uid: "r097WikN8kfJbZkWlZItqw7wMx73"
      },
      userInfo: { isLoaded: false, isEmpty: true },
      profile: { 456: "$56" }
      //uid: "r097WikN8kfJbZkWlZItqw7wMx73"
    },
    auth: {
      contributionDetails: { food: "foodName" }
    }
    // {"date":"2019-10-15","auth":{"isLoaded":false,"isEmpty":true},"userInfo":{"isLoaded":false,"isEmpty":true}}
  };
  const testReducer = (state = initState, action) => {
    return state;
  };

  let store = createStore(testReducer);
  const infoWithRudexAuth = renderer
    .create(
      <Provider
        // store={store}
        store={store}
      >
        <BrowserRouter>
          <Info />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(infoWithRudexAuth).toMatchSnapshot();
});
