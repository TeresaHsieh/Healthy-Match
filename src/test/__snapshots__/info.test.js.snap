// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Info 1`] = `null`;

exports[`Test Info with UID 1`] = `
<div>
  <header
    className="header"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        className="logo"
        src="test-file-stub"
      />
    </a>
    <div>
      <button
        className="hamburger-button"
        onClick={[Function]}
      >
        <div
          className="hamburger-line"
        />
        <div
          className="hamburger-line"
        />
        <div
          className="hamburger-line"
        />
      </button>
      <div />
    </div>
  </header>
  <div
    className="main-info"
  >
    <div
      className="backgroundSignUpRedirectBox"
    >
      <div
        className="signUpRedirectBox"
      >
        <div
          className="signUpRedirectBoxDetails"
        >
          <img
            className="love-match"
            src="test-file-stub"
          />
          <p>
            歡迎來到 Healthy Match！
          </p>
          <span>
            開始使用前，
            <br />
            請將基本資料填寫完畢，以利營養紀錄更準確喔！
            <br />
            也順便幫你的麻吉取個動聽的名字吧～～～
          </span>
        </div>
         
        <button
          onClick={[Function]}
        >
          <span>
            我知道了！
          </span>
        </button>
      </div>
    </div>
    <div
      className="more-info"
    >
      <div
        className="more-info-basic"
      >
        <div
          className="titleTitle"
        >
          <div
            className="titlewords"
          >
            用戶資料
          </div>
          <div
            className="divideLine"
          />
          <img
            className="basicInfoEdit"
            onClick={[Function]}
            src="/test-file-stub"
          />
        </div>
        <div
          className="userName"
        >
          <div>
            用戶姓名
          </div>
          <input
            disabled="disabled"
            name="Name"
            onChange={[Function]}
            placeholder="輸入姓名"
          />
        </div>
        <div
          className="userEmail"
        >
          <div>
            用戶信箱
          </div>
          <input
            disabled="disabled"
            name="Email"
            onChange={[Function]}
            placeholder="輸入信箱"
          />
        </div>
        <div
          className="userAge"
        >
          <div>
            用戶年齡
          </div>
          <input
            disabled="disabled"
            name="Age"
            onChange={[Function]}
            placeholder="輸入年齡"
          />
          歲
        </div>
        <div
          className="userSexual"
        >
          <div>
            用戶性別
          </div>
          <input
            disabled="disabled"
            name="Sexual"
            onChange={[Function]}
            placeholder="輸入性別"
          />
        </div>
        <div
          className="userHeight"
        >
          <div>
            用戶身高
          </div>
          <input
            disabled="disabled"
            name="Height"
            onChange={[Function]}
            placeholder="輸入身高"
          />
          公分
        </div>
        <div
          className="userWeight"
        >
          <div>
            用戶體重
          </div>
          <input
            disabled="disabled"
            name="Weight"
            onChange={[Function]}
            placeholder="輸入體重"
          />
          公斤
        </div>
        <div
          className="userMatchName"
        >
          <div>
            麻吉名字
          </div>
          <input
            disabled="disabled"
            name="MatchName"
            onChange={[Function]}
            placeholder="輸入麻吉名字"
          />
        </div>
        <div
          className="userMatchPic"
        >
          <div>
            麻吉照片
          </div>
          <img
            className="ChefMatch2"
          />
        </div>
      </div>
      <div
        className="more-info-match"
      >
        <div
          className="more-info-match"
        >
          <div
            className="titleTitle"
          >
            <div
              className="titlewords"
            >
              營養素貢獻紀錄
            </div>
            <div
              className="divideLine"
            />
          </div>
          <div
            className="userMatchName"
          >
            <div>
              目前尚未有紀錄
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
